### ISSUE TITLE
[TypeScript] ParseError when using class syntax inside svelte components

### ISSUE THREAD

**Issue by idiotWu** [sveltejs/eslint-plugin-svelte3#137]
## The Problem

Using `class` syntax inside svelte components will cause a `ParseError` in eslint, for example:

App.svelte:
```svelte
<script lang="ts">
  class Test {
    static value = 'test';
  }
</script>
```

eslint outputs:
```
.../App.svelte
  3:18  error  Unexpected token  ParseError

✖ 1 problem (1 error, 0 warnings)
```

## Steps to Reproduce

1. clone this repo: https://github.com/idiotWu/svelte-eslint-test
2. run `npm install`
3. run `npm run lint`

**Comment by Conduitry** [comment]
This syntax is unsupported in ESLint itself in regular JS files - https://github.com/eslint/eslint/issues/15016 - Once ESLint supports this and you've upgraded, this should work without any further changes on Svelte's side.

**Comment by idiotWu** [comment]
I think this problem is caused by the `target: ts.ScriptTarget.ESNext` option below:

https://github.com/sveltejs/eslint-plugin-svelte3/blob/e68ea6525bf25bc4baab5790179bc69cac299a18/src/preprocess.js#L238-L242

TypeScript (4.4.3 in my environment) will preserve class field declarations when targeting ESNext, for example,

```ts
// source.ts
export class Test {
  static value = 'test';
  this_breaks_too: number;
}
```

yields

```js
// source.js
export class Test {
    static value = 'test';
    this_breaks_too;
}
``` 

which breaks eslint. Is there any reason that this plugin uses the above compiler options instead of a user-defined `tsconfig.json`?

**Comment by iva2k** [comment]
@Conduitry: The issue you linked in eslint https://github.com/eslint/eslint/issues/15016 which substantiated closing this one is completely unrelated, it talks only about `static` variables. It has been fixed in eslint but it has no bearing here - this issue remains broken.

This issue at hand seems to fail to parse into valid code, so it is not an eslint rule that fails, but it throws "error  Unexpected token  ParseError", which does not seem to be originating in eslint, but perhaps in svelte compiler after typescript transpiler. All samples below fail:

```svelte
<script lang="ts">
  class Test1 {
    value = 'test';
    this_breaks_too: number;
  }
</script>
```

```svelte
<script>
  class Test2 {
    value = 'test';
    this_breaks_too = 0;
  }
</script>
```

Per https://github.com/sveltejs/svelte/issues/6900, setting `lib: ["esnext"]` in "tsconfig.json" makes svelte compiler produce invalid code. That is what this module does on L239: https://github.com/sveltejs/eslint-plugin-svelte3/blob/e68ea6525bf25bc4baab5790179bc69cac299a18/src/preprocess.js#L238-L242

Setting `lib ["es2020"]` fixes the problem in https://github.com/sveltejs/svelte/issues/6900.

I also patched L239 and it fixed the problem:

```js
function compile_code(text, compiler, processor_options) {
	const ts = processor_options.typescript;
	if (!ts) {
		return compiler.compile(text, { generate: false, ...processor_options.compiler_options });
	} else {
		const ts_options = {
			reportDiagnostics: false,
			compilerOptions: {
-				target: ts.ScriptTarget.ESNext,
+				target: ts.ScriptTarget.ES2020,
```

```bash
  10:9  warning  'Test1' is defined but never used  @typescript-eslint/no-unused-vars
  14:9  warning  'Test2' is defined but never used  @typescript-eslint/no-unused-vars
```

@Conduitry: please reopen this issue. And as a fix allow compiler options from tsconfig.json to pass into typescript compiler, or at least allow compiler options in `.eslintrc`, so the lib, target and module can be set.

**Comment by jescalan** [comment]
Just want to note that this is still an issue for me - a basic class property annotation inside a script block with `lang="ts"` will still emit the following warning, though it will compile without issue as long as you add `"target": "ES2021"` to your typescript config.

```
Unexpected token
If you expect this syntax to work, here are some suggestions: 
If you use typescript with `svelte-preprocess`, did you add `lang="ts"` to your `script` tag? 
Did you setup a `svelte.config.js`? 
See https://github.com/sveltejs/language-tools/tree/master/docs#using-with-preprocessors for more info.svelte(parse-error)
```

**Comment by knd775** [comment]
Yep. It even happens in a fresh SvelteKit demo app.

```bash
pnpm create svelte@latest myapp # Select demo app + TS + ESLint
cd myapp
eslint .
```

results in:
```
~\myapp\src\routes\Counter.svelte
  10:19  error  Unexpected token  ParseError

~\myapp\src\routes\sverdle\+page.svelte
  4:14  error  Unexpected token  ParseError

✖ 2 problems (2 errors, 0 warnings)
```
